subprojects/junit5/src/main/java/org/mockito/junit5/MockitoExtension.java

```diff
@@ -0,0 +1,79 @@
+package org.mockito.junit5;
+
+
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.mockito.MockitoSession;
+import org.mockito.quality.Strictness;
+
+import java.lang.reflect.AnnotatedElement;
+import java.util.Optional;
+
+import static java.util.Optional.empty;
+import static java.util.Optional.ofNullable;
+import static org.mockito.Mockito.mockitoSession;
+import static org.mockito.quality.Strictness.WARN;
+
+public class MockitoExtension implements BeforeEachCallback, AfterEachCallback {
+
+    private MockitoSession session;
```

Comment:
JUnit does not seem to guarantee how to handle Extension instances, so I think that it is better to use `ExtensionContext#getStore(Namespace)` rather than holding the MockitoSession instance in the field.

Reply:
fixed